### PR TITLE
chore: added code standardisation

### PR DESCRIPTION
**Pull request by elijah0kello**
# Changes in this PR .
- Adds eslint
- Adds commit message linting 
- Adds tests for functionality
- Adds precommit hooks
- Removes unused dependencies

### CODE REVIEW

File: .testcaferc.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (geka-evk) [inline .testcaferc.js:1]: What for this file?

File: tests/tokenRegistration.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import TokenRegistrationPageModel from "./TokenRegistrationPageModel";
+import {Selector} from "testcafe";
```
Comment (geka-evk) [inline tests/tokenRegistration.test.ts]: Why don't we use Jest?  AFAIK, Jest is default test framework in Mojaloop

File: package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "benefiaciary-registration-portal",
```
Comment (geka-evk) [inline package.json]: Let's rename this package to `benefiaciary-registration-portal-frontend` or `benefiaciary-registration-portal-ui`

File: package.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "start": "set PORT=3007 && react-app-rewired start",
```
Comment (geka-evk) [inline package.json]: Is it necessary to hardcode port like this?  What if a developer needs to run UI on another port?

File: src/types/types.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export type TtokenData =  {
```
Comment (geka-evk) [inline src/types/types.ts:1]: typo?

File: tests/tokenRegistration.test.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+});
+
+test("Pressing the submit button", async t => {
+    await t
+        .click(page.paymentTypeDropdown)
+        .click(payeeIdTypeOption.withText("MSISDN"))
+        .typeText(page.payeeId,"8394U3434")
+        .click(page.registerButton)
+        .wait(1000)
+        .expect(Selector("#errorDiv").exists).eql(true);
```
Comment (geka-evk) [inline tests/tokenRegistration.test.ts]: What are we testing here: error case OR happy path?

File: .env [hunk 7]
```diff
@@ -0,0 +1,5 @@
+REACT_APP__API_URL = http://localhost:8081
+REACT_APP_PAYMENT_TYPES=[{"type":"MSISDN","name":"Mobile Number"},{"type":"ACCOUNT_NO","name":"Account Number"},{"type":"EMAIL","name":"Email"},{"type":"PERSONAL_ID","name":"Personal ID"},{"type":"BUSINESS","name":"Business"},{"type":"DEVICE","name":"Device"},{"type":"ACCOUNT_ID","name":"Account ID"},{"type":"IBAN","name":"IBAN"},{"type":"ALIAS","name":"Alias"}]
+REACT_APP_VM_IP=3.254.52.179
+REACT_APP_CLIENT_ID='XaOVhjFTX_H8UiZf-O1TuV4ChixshdO8RqghtA_cRUM'
+REACT_APP_ESIGNET_BASE_URL=esignet.collab.mosip.net
```
Comment (geka-evk) [inline .env]: How do we use these env vars? is it for testing, local development or production?

File: src/types/types.ts (hunk 5 above)
Comment (elijah0kello) [reply src/types/types.ts:1]: Not a typo. It was intentional to put the T there

File: tests/tokenRegistration.test.ts (hunk 2 above)
Comment (elijah0kello) [reply tests/tokenRegistration.test.ts]: Testcafe is the test framework that was recommended to me to use for testing this.

File: package.json (hunk 3 above)
Comment (elijah0kello) [reply package.json]: Sure

File: package.json (hunk 4 above)
Comment (elijah0kello) [reply package.json]: I will address this.

File: .env (hunk 7 above)
Comment (elijah0kello) [reply .env]: They are used in the code. I just removed them from where they are hard coded and put them in the env file.

File: .testcaferc.js (hunk 1 above)
Comment (elijah0kello) [reply .testcaferc.js:1]: Testcafe configuration file

File: .env (hunk 7 above)
Comment (elijah0kello) [reply .env]: They are for production use

File: tests/tokenRegistration.test.ts (hunk 6 above)
Comment (elijah0kello) [reply tests/tokenRegistration.test.ts]: Let me make this test more robust to test happy path

File: .env (hunk 7 above)
Comment (geka-evk) [reply .env]: The issue I see, at least, with these values being added to main branch of source code:
```
REACT_APP_VM_IP=3.254.52.179
REACT_APP_CLIENT_ID='XaOVhjFTX_H8UiZf-O1TuV4ChixshdO8RqghtA_cRUM'
```
There's no more that VM, and thus, no that IP. Also I'm not sure it's a good idea to commit client ID to git

File: src/TokenRegistrationPage.tsx [hunk 8]
```diff
@@ -19,17 +20,18 @@ const TokenRegistrationPage: React.FC = () => {
...
-  const postDataAndReturnToken = async (requestData: any) => {
+  const postDataAndReturnToken = async (requestData: unknown) => {
     try {
-      const apiUrl = process.env.API_URL
-
-      if (!apiUrl) {
-        console.error('One or more required environment variables are not defined');        
-      }
+      console.log('postDataAndReturnToken requestData:', requestData);
+      // const apiUrl = process.env.API_URL
```
Comment (geka-evk) [inline src/TokenRegistrationPage.tsx]: I think, we can remove this commented code, right?

File: .env (hunk 7 above)
Comment (elijah0kello) [reply .env]: I agree. Although they were already committed but maybe what needs to be done is commit it in a .env.example

Comment (elijah0kello) [reply .env]: Let me address the concern

File: src/types/types.ts (hunk 5 above)
Comment (geka-evk) [reply src/types/types.ts:1]: Got it, but I'd suggest in this case using such naming: `TTokenData`

File: src/TokenRegistrationPage.tsx (hunk 8 above)
Comment (elijah0kello) [reply src/TokenRegistrationPage.tsx]: Yes let me also address this.

**Review by geka-evk — APPROVED** [review]

**Pull request merged**